### PR TITLE
DOCS-8624-Scripting-studio-edits-kt

### PR DESCRIPTION
**Pull request by kevintroller**
Scripting 2.0 Studio Configuration doc

### CODE REVIEW

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 1]
```diff
@@ -0,0 +1,234 @@
+= Use Anypoint Studio to Configure Scripting Module - Mule 4
+
+Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.
+
+To add and configure the module in Studio:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
To add and configure Scripting Module in Studio:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+. <<create-mule-project,Create a Mule project>>.
+. <<add-connector-to-project,Add the module to your Mule project>>.
+. <<configure-input-source,Configure an input source for the module's flow>>.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. <<configure-input-source,Configure an input source for the module flow>>.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+. <<add-connector-operation,Add Execute operation to the flow>>.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. <<add-connector-operation,Add the Execute operation to the flow>>.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+. <<configure-other-fields,Configure other module fields>>.
+
+
+If you are new to configuring connectors in Studio, see xref:connectors::introduction/intro-config-use-studio.adoc[Use Studio to Configure a Connector]. If, after reading this topic, you need additional information about the module fields, see the xref:scripting-reference.adoc[Scripting Module Reference].
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
If you are new to configuring connectors in Studio, see Use Studio to Configure a Connector. If, after reading this topic, you need additional information about the module fields, see Scripting Module Reference.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+. Enter a name for your Mule project and click *Finish*.
+
+
+[[add-connector-to-project]]
+== Add the Module to Your Mule Project
+
+Add Scripting Module to a Mule project to automatically populate the XML code with the module's namespace and schema location, as well as add the required dependencies to the project's `pom.xml` file:
+
+. In the *Mule Palette* view, click *(X) Search in Exchange*.
+. In *Add Modules to Project*, type `Scripting Module` in the search field.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. In *Add Modules to Project*, type `scripting module` in the search field.
```
For a search, you don't need caps.

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+. Click *Finish*.
+
+Adding a module to a Mule project in Studio does not make that module available to other projects in your Studio workspace.
+
+
+[[configure-input-source]]
+== Configure an Input Source
+
+An input source initiates a flow when a specified condition is met.
+You can configure one of these input sources to use with the Scripting Module:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
You can configure one of these input sources to use with Scripting Module:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+. Specify a value for the *Path* field.
+. Click the plus sign (+) next to the *Connector configuration* field to configure a global element that can be used by all instances of `HTTP Listener` in the app.
+. On the *General* tab, specify connection information.
+. On the *TLS* tab, optionally specify TLS information.
+. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
+. Click *Test Connection* to confirm that Mule can connect with the specific server.
+. Click *OK*.
+
+[[add-connector-operation]]
+== Add Execute Operation to the Flow
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
== Add the Execute Operation to the Flow
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+
+When you add a module operation to your flow, you immediately define a specific operation for that module to perform.
+
+To add the Execute operation for Scripting Module, follow these steps:
+
+. In the *Mule Palette* view, select *Scripting Module* and then select the Execute operation.
+. Drag the *Execute* operation onto the Studio canvas to the right of the input source.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]: Is "Execute" the label or not?
These two sentences use it in different ways, which I don't think you intend?

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+. Click the *Execute* operation to open the Execute configuration tab.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. Click *Execute* to open the Execute configuration tab.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+. Click *OK*.
+
+[[add-connector-operation]]
+== Add Execute Operation to the Flow
+
+When you add a module operation to your flow, you immediately define a specific operation for that module to perform.
+
+To add the Execute operation for Scripting Module, follow these steps:
+
+. In the *Mule Palette* view, select *Scripting Module* and then select the Execute operation.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. In the *Mule Palette* view, select *Scripting Module* and then select *Execute*.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+. Drag the *Execute* operation onto the Studio canvas to the right of the input source.
+. Click the *Execute* operation to open the Execute configuration tab.
+. Use one of the following two options to install and configure a `jsr-223` compliant engine: +
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. Use one of the following two options to install and configure a `jsr-223`-compliant engine: +
```
"compliant" is always preceded by a hyphen if the source of compliance comes first:

     a Mule-compliant app
     an app that is compliant with Mule

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+
+* In the *Required Libraries* section, click *Configure...* and select one of the following options to search and install the engine: +
+
+** Add recommended libraries +
+Install Groovy engine as recommended. This option only appears if you did not previously install Groovy.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
Install the Groovy engine as recommended. This option appears only if you did not previously install Groovy.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+** Use local file +
+Browse a local file for the required engine library and install it.
+** Add Maven dependency +
+Browse the engine dependency and install it. +
+
+* In the `pom.xml` file, manually add the engine in both of these sections:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]: Are you adding the engine to the TAG or to the SECTION in the following two sentences?

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+** `<sharedLibraries>` tags under the `<configuration>` section
+** `<dependency>` tags under the `<dependencies>` section
+
+[start=5]
+. In the *General* section of the *Execute* tab, click the refresh button to update the engine values list for the operation.
+. In *Engine*, select the installed engine.
+. Provide your script code in the *Code* text box. +
+Additionally, you can load script code from an external file by using a xref:mule-runtime::configuring-properties.adoc#_file_properties[file configuration property].
+
+In Studio, configuring Jython engine looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In Studio, configuring Jython engine looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+
+.Installing engine using Configure button
+image::scripting-configure-button.png[Installing engine using Configure button]
+
+In the `pom.xml` file, manually adding Jython engine looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In the `pom.xml` file, manually adding the Jython engine looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+<dependencies>
+  <dependency>
+    <groupId>org.python</groupId>
+    <artifactId>jython-standalone</artifactId>
+   <version>2.7.2</version>
+  </dependency>
+</dependencies>
+----
+
+In Studio, the Scripting Module configuration with script code looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In Studio, the Scripting Module configuration with script code looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+
+.Scripting Module configuration with script code
+image::scripting-studio-flow.png[Scripting Module configuration with script code]
+
+In the *Configuration XML* editor, the XML looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In the *Configuration XML* editor, the XML looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+      def factorial(n):
+          if n == 0: return 1
+	  return n * factorial(n-1)
+
+      result = factorial(10)
+    </scripting:code>
+</scripting:execute>
+----
+
+In Studio, the loaded script using an external file looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In Studio, the loaded script using an external file looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 19]
```diff
@@ -0,0 +1,234 @@
...
+
+.Script code using an external file
+image::scripting-studio-file-config.png[Script code using an external file]
+
+In the *Configuration XML* editor, place the script in between `<scripting:code>` tag as follow:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In the *Configuration XML* editor, place the script in between `<scripting:code>` tags, as follows:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+----
+<scripting:execute engine="python">
+    <scripting:code >${file::script.py}</scripting:code>
+</scripting:execute>
+----
+
+[[configure-other-fields]]
+== Configure Other Module Fields
+
+You can configure other additional fields for the Execute operation, such as Parameters, Target Variable, and Target Value.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
You can configure other additional fields for the Execute operation, such as *Parameters*, *Target Variable*, and *Target Value*.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 21]
```diff
@@ -0,0 +1,234 @@
...
+
+
+=== Parameters Parameter
+
+Use the *Parameters* parameter to define input values for the script to use through DataWeave. For the DataWeave expression to work correctly, you must combine the output types of the parameters, in which keys are strings and values are any object. +
+Reference the parameters by name to use them as binding variables, for example: +
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
Use the *Parameters* parameter to define input values for the script to use through DataWeave. For the DataWeave expression to work correctly, you must combine the output types of the parameters, in which keys are strings and values are any object. Reference the parameters by name to use them as binding variables: +
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 22]
```diff
@@ -0,0 +1,234 @@
...
+
+`factorial(initialValue + int(payload))`
+
+To configure this parameter for the Scripting Module:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
To configure this parameter for Scripting Module:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 23]
```diff
@@ -0,0 +1,234 @@
...
+
+. Select the name of the connector in the Studio canvas.
+. On the *General* section, add parameter values in the *Parameters* field.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
. In the *General* section, add parameter values in the *Parameters* field.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 24]
```diff
@@ -0,0 +1,234 @@
...
+
+In Studio, the Parameters configuration looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In Studio, the parameters configuration looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 25]
```diff
@@ -0,0 +1,234 @@
...
+
+.Script using parameters
+image::scripting-studio-parameters.png[Script using parameters]
+
+In the *Configuration XML* editor, the XML looks as follows :
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In the *Configuration XML* editor, the XML looks like this:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 26]
```diff
@@ -0,0 +1,234 @@
...
+result = factorial(initialValue + int(payload))</scripting:code>
+	<scripting:parameters ><![CDATA[#[{
+        initialValue: 3
+    }]]]></scripting:parameters>
+</scripting:execute>
+----
+
+=== Target Variable and Target Value
+
+Use the *Target Variable* parameter to define the name of a variable in which to place the operation’s output, and the *Target Value* parameter to define an expression to evaluate against the operation’s output. The outcome of that expression will be stored in the target variable.
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
Use the *Target Variable* parameter to define the name of a variable in which to place the operation’s output, and the *Target Value* parameter to define an expression to evaluate against the operation’s output. The outcome of that expression is stored in the target variable.
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 27]
```diff
@@ -0,0 +1,234 @@
...
+
+To configure these parameters for the Scripting Module:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
To configure these parameters for Scripting Module:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 28]
```diff
@@ -0,0 +1,234 @@
...
+
+. Select the name of the connector in the Studio canvas.
+. On the *Advanced* tab, define *Target Variable* and *Target Value* as the xref:mule-runtime::target-variables.adoc[target] of the scripting execution.
+
+In Studio, the Target Variable and Target Value configuration looks as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In Studio, the target variable and target value configuration looks as follows:
```

File: scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc [hunk 29]
```diff
@@ -0,0 +1,234 @@
...
+
+.Script Target Variable and Target Value
+image::scripting-studio-target.png[Advanced settings]
+
+In the *Configuration XML* editor, the XML looks similar as follows:
```
Comment (kbostrom) [inline scripting/2.0/modules/ROOT/pages/edits/scripting-module-studio.adoc]:
```suggestion
In the *Configuration XML* editor, the XML looks similar to the following:
```

**Review by kbostrom — DISMISSED** [review]
Edited

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**